GoogleDataTransportCCTSupport/GDTCCTLibrary/GDTFLLUploader.m

```diff
@@ -309,6 +314,26 @@ - (nonnull NSData *)constructRequestProtoFromPackage:(GDTCORUploadPackage *)pack
   return data ? data : [[NSData alloc] init];
 }
 
+/** Constructs a request to FLL given a URL and request body data.
+ *
+ * @param URL The URL to send the request to.
+ * @param data The request body data.
+ * @return A new NSURLRequest ready to be sent to FLL.
+ */
+- (NSURLRequest *)constructRequestWithURL:(NSURL *)URL data:(NSData *)data {
+  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
+  [request setValue:[self defaultAPIKey] forHTTPHeaderField:@"X-Goog-Api-Key"];
+  [request setValue:@"application/x-protobuf" forHTTPHeaderField:@"Content-Type"];
+  [request setValue:@"gzip" forHTTPHeaderField:@"Content-Encoding"];
+  [request setValue:@"gzip" forHTTPHeaderField:@"Accept-Encoding"];
+  NSString *userAgent = [NSString stringWithFormat:@"datatransport/%@ fllsupport/%@ apple/",
+                                                   kGDTCORVersion, kGDTCCTSupportSDKVersion];
+  [request setValue:userAgent forHTTPHeaderField:@"User-Agent"];
+  request.HTTPMethod = @"POST";
+  [request setHTTPBody:data];
```

Comment:
BTW, I suspect it may solve the issue even for 302 and the delegate method will be unnecessary. Do you think we can verify it?

Reply:
I just finished writing a unit test with a real HTTP server. With the delegate method implemented, it passes all redirects. Without the delegate method implemented, it fails on 301 and 302 redirects and passes on 307.

![image](https://user-images.githubusercontent.com/1082754/68904161-23c22600-06f2-11ea-8891-6d944d8d7eda.png)